### PR TITLE
[async_backing] Bump ParachainHost to api version 10 on polkadot

### PR DESCRIPTION
**Pull request by alexggh**
This will enable async-backing subsystems on polkadot, the relaychain would work in legacy mode where only candiddates built on top of pervious relay chain are activated.

## Notes
- Had to bring an unrelated change as well `minimum_backing_votes` because that was the v6, we can't skip versions, the value of that configuration on polkadot is 2, so that's what is going to be used after this runtime update. Changes have been running on kusama.
- `disabled_validators`, `node_features`, `approval_voting_params` are not related with async-backing, but given they are low risk, we decided to include them as well, see comments.

## Known risks

Async backing subsytems is a major change in the way collator-protocol and  the backing work, so there are still some unknowns if this is completely bug free. It has been running on kusama for a month already, but not without issues:
 - Validators that did not upgrade to compatible versions will not be able to participate in backing, so if enough of those are randomly picked that group won't be able to back anything. With backing_group_size = 5 and minimum_backing_votes = 2, 10% validator not upgraded,  that chance is about 2.5%.

 - Additionally, same un-upgraded groups won't be able to include the backing votes on chain when they author blocks, so 10% of the blocks won't back any candidates.

 - We are still not sure if item 2) from here https://github.com/paritytech/polkadot-sdk/issues/3314#issuecomment-1953542045 is caused by async backing, the observable issue is sometimes after restart/upgrade some validators are getting 0 rewards and apparently they are not backing anything.

### CODE REVIEW

File: integration-tests/emulated/chains/relays/polkadot/src/lib.rs [hunk 1]
```diff
@@ -21,7 +21,7 @@ use emulated_integration_tests_common::{
 	impl_hrmp_channels_helpers_for_relay_chain, impl_send_transact_helpers_for_relay_chain,
 	xcm_emulator::decl_test_relay_chains,
 };
-use polkadot_primitives::runtime_api::runtime_decl_for_parachain_host::ParachainHostV5;
```
Comment (eskimor) [inline integration-tests/emulated/chains/relays/polkadot/src/lib.rs:24]: I think there is no strong reason not to bump to version 10 right away? Are there any other node features which are enabled via runtime API version?

Also with the release on Polkadot, we should be able to bump the base version [here](https://github.com/paritytech/polkadot-sdk/blob/a84dd0dba58d51503b8942360aa4fb30a5a96af5/polkadot/primitives/src/runtime_api.rs#L131).

Comment (eskimor) [reply integration-tests/emulated/chains/relays/polkadot/src/lib.rs:24]: Put differently, I believe the blocker for bumping the API was asynchronous backing. If that blocker is removed, I am not aware of anything holding back the remaining versions.

Comment (alexggh) [reply integration-tests/emulated/chains/relays/polkadot/src/lib.rs:24]:
```
		/// Returns a list of all disabled validators at the given block.
		#[api_version(8)]
		fn disabled_validators() -> Vec<ValidatorIndex>;

		/***** Added in v9 *****/

		/// Get node features.
		/// This is a staging method! Do not use on production runtimes!
		#[api_version(9)]
		fn node_features() -> vstaging::NodeFeatures;

		/***** Added in v10 *****/
		/// Approval voting configuration parameters
		#[api_version(10)]
		fn approval_voting_params() -> ApprovalVotingParams;
```

These are the other, node_features and approval_voting_params are 0 risk, for `disabled_validators` I will let @ordian and @tdimitrov answer.

Comment (eskimor) [reply integration-tests/emulated/chains/relays/polkadot/src/lib.rs:24]: disabled validators should also not do anything. Just exposing the API for later use.

Comment (alexggh) [reply integration-tests/emulated/chains/relays/polkadot/src/lib.rs:24]: Done.

**Review by bkchr — APPROVED** [review]

File: relay/polkadot/src/lib.rs [hunk 2]
```diff
@@ -2101,6 +2106,30 @@ sp_api::impl_runtime_apis! {
...
+		fn disabled_validators() -> Vec<ValidatorIndex> {
+			parachains_vstaging_api_impl::disabled_validators::<Runtime>()
+		}
+
+		fn node_features() -> NodeFeatures {
+			parachains_vstaging_api_impl::node_features::<Runtime>()
+		}
+
+		fn approval_voting_params() -> ApprovalVotingParams {
+			parachains_vstaging_api_impl::approval_voting_params::<Runtime>()
```
Comment (ordian) [inline relay/polkadot/src/lib.rs:2131]: not a blocker as we can do it as follow-up cleanup PR, but we should not use use vstaging runtime impls on Polkadot and instead stabilize them once tested on Kusama

File: relay/polkadot/src/lib.rs [hunk 3]
```diff
@@ -2101,6 +2106,30 @@ sp_api::impl_runtime_apis! {
...
+		}
```
Comment (ordian) [inline relay/polkadot/src/lib.rs:2132]: not a blocker as we can do it as follow-up cleanup PR, but we should not use use vstaging runtime impls on Polkadot and instead stabilize them once tested on Kusama

**Review by ordian — APPROVED** [review]

**Review by eskimor — APPROVED** [review]

File: relay/polkadot/src/lib.rs (hunk 3 above)
Comment (alexggh) [reply relay/polkadot/src/lib.rs:2132]: Ok, then I'll clean this up with a follow up PR.

**Comment by alexggh** [comment]
Bot fails to merge it, it seems another required migration is failing @bkontur @bkchr thoughts on how to get this merged ?

**Comment by bkontur** [comment]
> Bot fails to merge it, it seems another required migration is failing @bkontur @bkchr thoughts on how to get this merged ?

e.g. push empty commit and wait for CI again or find someone who can restart that job: https://github.com/polkadot-fellows/runtimes/actions/runs/8248792955/job/22560279689?pr=222, maybe @ggwpez could help?

**Pull request merged**